macros/contexts/contextUnits.pl

```diff
@@ -0,0 +1,2076 @@
+denominator in parentheses, as in
+
+    $n = Compute("3 kg/m/s");
+    $n = Compute("3 kg/(m s)");
+    $n = Compute("3 kg per meter per second");
+
+Units can be preceded by formulas as well as numbers.  For example
+
+    $f = Compute("2x meters");
+
+makes C<$f> be a Formula returning a Number-with-Unit.  Note, however,
+that since the space before the unit has the same precedence as
+multiplication (just as it does within a formula), if the expression
+before the unit includes addition, you need to enclose it in parentheses:
+
+    $n = Compute("(1+4) meters");
+    $f = Compute("(1+2x) meters");
+
+Using C<Compute()> is not the only way to produce a number or formula
+with units; there are also constructor functions that are sometimes
+useful when writing a problem involving units.
+
+    $n = NumberWithUnits(3, "m/s");
+    $f = FormulaWithUnits("1+2x", "meters");
+
+These are most useful when the numeric part is the result of a
+computation or a value held in a variable:
+
+    $n = NumberWithUnits(random(1,5), "m");
+
+Since units are themselves MathObjects, you can work with units
+without a preceding number.  These can be created through C<Compute()>
+just as with other MathObject, or you can use the C<Unit()> constructor.
+
+    $u = Compute("meters per second per second");
+    $u = Unit("m/s^2");
+
+This allows you to ask a student to say what units should be used for
+a particular setting, without the need for a quntity.
+
+
+=head2 Working with numbers with units
+
+Because units and numbers with units are full-fledged MathObjects, you
+can do computations with them, just as with other MathObejcts.  For
+example, you can do
+
+    $n = Compute("3 m + 10 cm");
+
+to get the equivalent of C<3.1 m>.  Similarly, you can do
+
+    $velocity = Compute("100 miles / (2 hours)");  # equals "50 mi/h"
+    $area = Compute("(5 m) * (3 m)");              # equals "15 m^2"
+
+to get numbers with compound units.
+
+As with other MathObjects, units and numbers with units can be
+combined using perl operations:
+
+    $distance = Compute("100 miles");
+    $time = Compute("2 hours");
+    $velocity = $distance / $time;  # equivalent to "50 miles/hour"
+
+    $m = Compute("m");
+    $s = Compute("s");
+    $a = 9.8 * $m / $s**2;
+
+    $x = Compute("x");
+    $f = (3 * $x**2 - 2) * $m;  # equivalent to Compute("(3x^2 - 2) m");
+
+The units objects provide functions for converting from one set of
+units to another (compatible) set via the C<toUnits()> and
+C<toBaseUnits()> methods.  For example:
+
+    $m = Compute("5 m");
+    $ft = $m->toUnits("ft");                 # returns "16.4042 ft"
+
+    $cm = Compute("5.21 m")->toUnits("cm");  # returns "521 cm"
+
+    $a = Compute("32 ft/s^2")->toBaseUnits;  # returns "9.7536 m/s^2"
+
+For a given number with units, you may wish to obtain the numeric
+portion or the units portion separatly.  This can be done using the
+C<number> and C<unit> methods:
+
+    $n = Compute("5 m");
```

Comment:
Should this be `$m = Compute("5 m"):`?  I think this is the variable referenced below.

The same thing for the `$n` on line 339.

Reply:
I'll fix in a separate commit.